test/hotspot/gtest/runtime/test_os.cpp

```diff
@@ -385,25 +394,36 @@ static address reserve_multiple(int num_stripes, size_t stripe_len) {
   const bool exec_supported = can_reserve_executable_memory();
 #endif
 
-  size_t total_range_len = num_stripes * stripe_len;
-  // Reserve a large contiguous area to get the address space...
-  address p = (address)os::reserve_memory(total_range_len);
-  EXPECT_NE(p, (address)NULL);
-  // .. release it...
-  EXPECT_TRUE(os::release_memory((char*)p, total_range_len));
-  // ... re-reserve in the same spot multiple areas...
-  for (int stripe = 0; stripe < num_stripes; stripe++) {
-    address q = p + (stripe * stripe_len);
-    // Commit, alternatingly with or without exec permission,
-    //  to prevent kernel from folding these mappings.
+  address p = NULL;
+  for (int tries = 0; tries < 256 && p == NULL; tries ++) {
+    size_t total_range_len = num_stripes * stripe_len;
+    // Reserve a large contiguous area to get the address space...
+    p = (address)os::reserve_memory(total_range_len);
+    EXPECT_NE(p, (address)NULL);
+    // .. release it...
+    EXPECT_TRUE(os::release_memory((char*)p, total_range_len));
+    // ... re-reserve in the same spot multiple areas...
+    for (int stripe = 0; stripe < num_stripes; stripe++) {
+      address q = p + (stripe * stripe_len);
+      // Commit, alternatingly with or without exec permission,
+      //  to prevent kernel from folding these mappings.
 #ifdef __APPLE__
-    const bool executable = exec_supported ? (stripe % 2 == 0) : false;
+      const bool executable = exec_supported ? (stripe % 2 == 0) : false;
 #else
-    const bool executable = stripe % 2 == 0;
+      const bool executable = stripe % 2 == 0;
 #endif
-    q = (address)os::attempt_reserve_memory_at((char*)q, stripe_len, executable);
-    EXPECT_NE(q, (address)NULL);
-    EXPECT_TRUE(os::commit_memory((char*)q, stripe_len, executable));
+      q = (address)os::attempt_reserve_memory_at((char*)q, stripe_len, executable);
+      if (q == NULL) {
+        // Someone grabbed that area concurrently. Cleanup, then retry.
+        tty->print_cr("reserve_multiple: retry (%d)...", stripe);
+        if (stripe > 0) {
```

Comment:
Nit: you don't need to check this as `carefully_release_multiple` is a no-op for `num_stripes==0`.